nvflare/edge/assessors/model_update.py

```diff
@@ -53,10 +56,154 @@ def __init__(
+                        f"expected: {self.device_manager.device_selection_size}. "
+                        f"Timeout in {remaining_time:.1f} seconds.",
+                    )
+                    self._last_status_log_time = current_time
+
+    def _start_timeout_tracking(self, fl_ctx: FLContext):
+        """Start timeout tracking using long-lived thread."""
+        if not self.timeout_thread_active:
+            # Start the long-lived timeout thread if not already running
+            self.timeout_future = self.thread_pool.submit(self._timeout_tracker, fl_ctx)
+            self.timeout_thread_active = True
+            self.log_debug(fl_ctx, f"Started device wait timeout tracking for {self.device_wait_timeout}s")
+
+        # Signal the thread to start monitoring
+        self.timeout_wait_event.set()
+
+    def _stop_timeout_tracking(self):
+        """Stop timeout tracking."""
+        # Clear the wait event to signal the thread to stop monitoring
+        self.timeout_wait_event.clear()
+
+    def _timeout_tracker(self, fl_ctx: FLContext):
+        """Long-lived timeout tracker that runs in thread pool."""
+        try:
+            while True:
+                # Wait for the event to be set (indicating we should start monitoring)
+                if not self.timeout_wait_event.wait(timeout=1.0):
+                    # Timeout on wait - check if we should exit the thread
+                    if self.should_stop_job:
+                        break
+                    continue
+
+                # Event was set, start monitoring timeout
+                if self.device_wait_start_time is None:
+                    # No timeout to monitor, clear event and continue
+                    self.timeout_wait_event.clear()
+                    continue
+
+                # Periodic logging during countdown to keep users informed
+                check_interval = 10.0  # Log every 10 seconds
+                elapsed = 0
+
+                while (
+                    self.device_wait_start_time is not None
+                    and not self.should_stop_job
+                    and elapsed < self.device_wait_timeout
+                    and self.timeout_wait_event.is_set()
+                ):
+                    time.sleep(min(check_interval, self.device_wait_timeout - elapsed))
+                    elapsed += check_interval
+
+                    # Check if we should stop monitoring
+                    if (
+                        self.device_wait_start_time is None
+                        or self.should_stop_job
+                        or not self.timeout_wait_event.is_set()
+                    ):
+                        break
+
+                    # Log periodic status update
+                    if elapsed < self.device_wait_timeout:
+                        remaining = self.device_wait_timeout - elapsed
+                        self.log_info(fl_ctx, f"Device wait countdown: {remaining:.1f} seconds remaining")
```

Comment:
instead of doing nested while loop, maybe we have time.sleep(1.0) for the outer while loop, and then we just use "if" statements here?